hydromt/components/mesh.py

```diff
@@ -1,40 +1,384 @@
 """Implementations for model mesh workloads."""
 
-import logging
 import os
 from os.path import dirname, isdir, join
 from pathlib import Path
-from typing import Dict, List, Optional, Union
+from typing import TYPE_CHECKING, Dict, List, Optional, Union
 
 import geopandas as gpd
 import pandas as pd
 import xarray as xr
 import xugrid as xu
 from pyproj import CRS
+from shapely.geometry import box
 
-from hydromt import workflows
+from hydromt import hydromt_step
+from hydromt.components.base import ModelComponent
 from hydromt.gis.raster import GEO_MAP_COORD
-from hydromt.models.model import Model
+from hydromt.io.readers import read_nc
+from hydromt.workflows.mesh import (
+    create_mesh2d,
+    mesh2d_from_raster_reclass,
+    mesh2d_from_rasterdataset,
+    rename_mesh,
+)
 
-__all__ = ["MeshModel"]
-logger = logging.getLogger(__name__)
+if TYPE_CHECKING:
+    from hydromt.models import Model
 
 
-class MeshMixin(object):
-    # placeholders
-    # We cannot initialize an empty xu.UgridDataArray
-    _API = {
-        "mesh": Union[xu.UgridDataArray, xu.UgridDataset],
-    }
+DEFAULT_FN = "mesh/mesh.nc"
 
-    def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
-        self._mesh = None
+__all__ = ["MeshComponent"]
 
-    ## general setup methods
-    def setup_mesh2d_from_rasterdataset(
+
+class MeshComponent(ModelComponent):
+    """ModelComponent class for mesh components."""
+
+    def __init__(self, model: "Model"):
+        super().__init__(model=model)
```

Comment:
We just discussed yesterday that we want to pass a default filename in the component __init__ and then use None in the argument for read and write. Can you update?
You can check out the TablesComponent PR